src/irmin/tree.ml

```diff
@@ -315,6 +315,10 @@ module Make (P: S.PRIVATE) = struct
       in
       P.Node.Val.v alist
 
+    let to_key t = match t.v with
+      | Key (_, k) | Both (_, k, _) -> k
+      | Map m -> P.Node.Key.digest P.Node.Val.t (export_map m)
```

Comment:
Not that this is general unsafe as the storage backend can choose a different serialisation that the default ont